### PR TITLE
Cleanup OpenInputDesktop/CloseDesktop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup User32.OpenInputDesktop
- Cleanup User32.CloseDesktop

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2024)

### CODE REVIEW

File: src/Common/src/NativeMethods.cs [hunk 1]
```diff
@@ -19,29 +19,6 @@ internal static class NativeMethods
...
-        SHGFI_SELECTED = 0x000010000,     // show icon in selected state
-        SHGFI_ATTR_SPECIFIED = 0x000020000,     // get only specified attributes
-        SHGFI_LARGEICON = 0x000000000,     // get large icon
-        SHGFI_SMALLICON = 0x000000001,     // get small icon
-        SHGFI_OPENICON = 0x000000002,     // get open icon
-        SHGFI_SHELLICONSIZE = 0x000000004,     // get shell size icon
-        SHGFI_PIDL = 0x000000008,     // pszPath is a pidl
-        SHGFI_USEFILEATTRIBUTES = 0x000000010,     // use passed dwFileAttribute
-        SHGFI_ADDOVERLAYS = 0x000000020,     // apply the appropriate overlays
-        SHGFI_OVERLAYINDEX = 0x000000040;     // Get the index of the overlay
```
Comment (RussKie) [inline src/Common/src/NativeMethods.cs]: Why were these deleted? They don't have anything to do with the nature of the PR, and are used by https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shgetfileinfow

File: src/Common/src/Interop/User32/Interop.DF.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [Flags]
+        public enum DF : uint
+        {
+            ALLOWOTHERACCOUNTHOOK = 0x0001
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.DF.cs]: I think we need to have a default value too
image

File: src/Common/src/Interop/User32/Interop.DESKTOP.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            READOBJECTS = 0x0001,
+            CREATEWINDOW = 0x0002,
+            CREATEMENU = 0x0004,
+            HOOKCONTROL = 0x0008,
+            JOURNALRECORD = 0x0010,
+            JOURNALPLAYBACK = 0x0020,
+            ENUMERATE = 0x0040,
+            WRITEOBJECTS = 0x0080,
+            SWITCHDESKTOP = 0x0100,
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.DESKTOP.cs]: I think the following should be added too for completeness:
image
https://docs.microsoft.com/en-us/windows/win32/winstation/desktop-security-and-access-rights

File: src/Common/src/Interop/User32/Interop.CloseDesktop.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(Libraries.User32, ExactSpelling = true)]
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.CloseDesktop.cs]: https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-closedesktop#return-value
```suggestion
        [DllImport(Libraries.User32, ExactSpelling = true,  SetLastError = true)]
```

**Comment by RussKie** [comment]
@msftbot merge in 24 hours

File: src/Common/src/Interop/Interop.ERROR.cs [hunk 5]
```diff
@@ -4,8 +4,9 @@
 
 internal partial class Interop
 {
-    internal static class Errors
+    internal static class ERROR
```
Comment (zsd4yr) [inline src/Common/src/Interop/Interop.ERROR.cs]: Any reason not to make this an enum called Win32ERROR ?

**Comment by RussKie** [comment]
@msftbot merge if @zsd4yr approves

File: src/Common/src/Interop/Interop.ERROR.cs (hunk 5 above)
Comment (hughbe) [reply src/Common/src/Interop/Interop.ERROR.cs]: Yes because thats not the interop name. All constants are name `ERROR_X` and its good to keep to convention

Comment (zsd4yr) [reply src/Common/src/Interop/Interop.ERROR.cs]: Doh!
